### PR TITLE
first commit for 3rd week

### PR DESCRIPTION
**Pull request by sardorboboshov**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,3 +1,6 @@
 PORT=3000
 JWT_SECRET=super_secret
-JWT_EXPIRES_IN=1h
\ No newline at end of file
+JWT_EXPIRES_IN=1h
+MONGODB_URL=mongodb://localhost:27017/admin
```
Comment (vokanred) [inline .env:4]: Do you use MONGODB_URL and DB_NAME?

File: app.js [hunk 2]
```diff
@@ -1,13 +1,8 @@
 /* eslint-disable prefer-destructuring */
-/* eslint-disable comma-dangle */
 require('dotenv').config();
 
 const express = require('express');
 const cors = require('cors');
-const passport = require('passport');
```
Comment (vokanred) [inline app.js:7]: Why did you remove passport?

**Review by vokanred — CHANGES_REQUESTED** [review]

File: models/users.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  id: {
+    type: Number,
+    required: true,
+    unique: true,
+  },
+  user_name: {
+    type: String,
+    required: true,
+  },
+  email: {
```
Comment (vokanred) [inline models/users.js:13]: You can add additional email validation here as well

File: controllers/authController.js [hunk 4]
```diff
@@ -26,21 +27,25 @@ exports.checkBodyRegister = (req, res, next) => {
...
-  );
-  if (ifUserExists) {
+  const ifUserExists = await User.find({ email: req.body.email });
+  if (Object.keys(ifUserExists).length !== 0) {
     return res.status(400).json({
       status: 'failed',
       message: 'user with that email exists already',
     });
   }
+  const emailIsValid = (email) => /[a-zA-Z0-9]@[a-zA-Z0-9]/.test(email);
```
Comment (vokanred) [inline controllers/authController.js:37]: You can use any validation packages

File: .env (hunk 1 above)
Comment (sardorboboshov) [reply .env:4]: Firstly I was using local mongoDB(and I was using MONGODB_URL and DB_NAME) and I switched to remote MongoDB and I forgot to delete those two variables in .env file. Should I delete them and open new pr or can I delete them in next hw?

File: app.js (hunk 2 above)
Comment (sardorboboshov) [reply app.js:7]: Using jwtCallback function(which was part of passport package) made some issues in application and it didnt run, so I removed that package. If it is required to use that package I will reinstall and use.

File: models/users.js (hunk 3 above)
Comment (sardorboboshov) [reply models/users.js:13]: Should I do validation in both: middleware and schema? Or using validation in only one of them is enough?

Comment (vokanred) [reply models/users.js:13]: Validation in model layer is safer

File: .env (hunk 1 above)
Comment (vokanred) [reply .env:4]: You can use connection string for a local db also.
Remove and create a new commit in the current branch (PR).

File: app.js (hunk 2 above)
Comment (vokanred) [reply app.js:7]: I thought  that it was your homework to use passport. Wasn't it?
You can ask how to fix such issues

File: controllers/authController.js (hunk 4 above)
Comment (sardorboboshov) [reply controllers/authController.js:37]: But why to use additional packages for small things that can be solved by js itself? Should I use package for validation?

Comment (vokanred) [reply controllers/authController.js:37]: Are you sure that you considered all possible cases?
For example I found this regex on stackoverflow.

(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])

Packages allow you not to think about such details

File: .env (hunk 1 above)
Comment (sardorboboshov) [reply .env]: I created new commit:
1)removed 2 variables from .env file
2)installed "validator" package and used them to validate email
3)added validation to the schema for email
4)reinstalled passport and used them and there are no issues right now.

File: controllers/authController.js (hunk 4 above)
Comment (vokanred) [reply controllers/authController.js]: Any news?

Comment (vokanred) [reply controllers/authController.js]: What about `const emailIsValid = (email) => /[a-zA-Z0-9]@[a-zA-Z0-9]/.test(email);`?
You haven't replaced it with validator

Comment (vokanred) [reply controllers/authController.js]: And request review again, because PR state now is Changes requested

Comment (sardorboboshov) [reply controllers/authController.js]: > 

As I understand I should create new pull request right?

**Review by sardorboboshov — COMMENTED** [review]
Commit after 1st review

**Review by vokanred — APPROVED** [review]

**Review by vokanred — APPROVED** [review]
Done. Pls merge

**Pull request merged**